### ISSUE TITLE
Rule proposal: Disallow unions with `unknown` or `any`

### ISSUE THREAD

**Issue by fregante** [typescript-eslint/typescript-eslint#4776]
### Description

From what I understand, `T | unknown` and `T | any` resolve to just `unknown` and `any` respectively, so I don't think they're particularly useful.

<img width="307" alt="Screen Shot 9" src="https://user-images.githubusercontent.com/1402241/161381218-8b22cb49-cb9d-4e4b-b686-7303294d99b5.png">

### Fail

```ts
type MaybeNumber = number | unknown;

function isNumber(value: number | any): value is number {
	return typeof value === 'number';
}
```

### Pass

```ts
type MaybeNumber = unknown;

function isNumber(value: any): value is number {
	return typeof value === 'number';
}
```

**Comment by bradzacher** [comment]
We already have such a rule - `no-redundant-type-constituents`

https://typescript-eslint.io/rules/no-redundant-type-constituents/

**Comment by fregante** [comment]
Indeed I had already asked for it but the search had no results with "unknown commenter:@me" 😩 

https://github.com/typescript-eslint/typescript-eslint/issues/3926

I remember we had already discussed this but I was starting to question my sanity.
